packages/market-matcher/src/EntityFetcher.ts

```diff
@@ -0,0 +1,89 @@
+import 'reflect-metadata';
+import { Agreement, Demand, Supply, PurchasableCertificate } from '@energyweb/market';
+import { Configuration, Currency } from '@energyweb/utils-general';
+import { inject, singleton } from 'tsyringe';
+import polly from 'polly-js';
+
+export interface IEntityFetcher {
+    getAgreementListLength(): Promise<number>;
+    getDemandListLength(): Promise<number>;
+    getSupplyListLength(): Promise<number>;
+    getCertificateListLength(): Promise<number>;
+
+    getAgreement(id: string, tries?: number): Promise<Agreement.Entity>;
+    getDemand(id: string, tries?: number): Promise<Demand.Entity>;
+    getSupply(id: string, tries?: number): Promise<Supply.Entity>;
+    getCertificate(id: string, tries?: number): Promise<PurchasableCertificate.Entity>;
+}
+
+@singleton()
+export class EntityFetcher implements IEntityFetcher {
+    constructor(@inject('config') private config: Configuration.Entity) {}
+
+    getAgreementListLength() {
+        return Agreement.getAgreementListLength(this.config);
+    }
+
+    getCertificateListLength() {
+        return PurchasableCertificate.getCertificateListLength(this.config);
+    }
+
+    getDemandListLength() {
+        return Demand.getDemandListLength(this.config);
+    }
+
+    getSupplyListLength() {
+        return Supply.getSupplyListLength(this.config);
+    }
+
+    getAgreement(id: string, tries = 1) {
```

Comment:
Previously the default was 10. FYI 1 retry means 1 failed and 1 immediately after. This is our "hack" for awaiting db and blockchain be in sync. I recommend setting it to higher value, perhaps 5-10.

Reply:
Set to 10